src/PeriodOfStayInput.js

```diff
@@ -7,29 +7,18 @@
         Model = require('./Model'),
         ReactIntl = require('react-intl'),
         FormattedMessage = ReactIntl.FormattedMessage,
-        IntlMixin = ReactIntl.IntlMixin,
-
-        enMessages = function () {
-            return {
-                'react-period-of-stay-input': {
-                    period: '{count, plural, =0 {Single day} =1 {1 night} other {# nights}}',
-                    checkInDay: 'Check-in day',
-                    checkOutDay: 'Check-out day'
-                }
-            };
-        };
+        IntlMixin = ReactIntl.IntlMixin;
 
     module.exports = React.createClass({
         mixins: [IntlMixin],
         propTypes: {
             environment: React.PropTypes.instanceOf(Environment),
             model: React.PropTypes.instanceOf(Model),
-            onChange: React.PropTypes.func
+            onChange: React.PropTypes.func,
+            messages: React.PropTypes.object.isRequired
```

Comment:
I doubt we should do this. That may prohibit the descent of messages down the components hierarchy, or, at least may be confusing for the users, who may think it's required to pass the messages prop explicitly.